### PR TITLE
LPS-120006 WIP Test timing storing segment user in session

### PR DESCRIPTION
**Pull request by darquesdev**
TTFB (Time to first byte)

MASTER:
100 segments 516,82ms
200 segments 1,07ms
500 segments 2,52ms

Segments user stored in cache used in SegmentsPreAction + SegmenteRoleContributor

First call
50 segments 359ms
100 segments (380 - 680) ms
200 segments (600 - 800) ms
500 segments (1.5 - 3.2) s

Rest of calls
50 segments (12 - 19)ms
100 segments (7 - 16) ms
200 segments (8 - 17) ms
500 segments (8 - 17)ms

### CODE REVIEW

**Comment by darquesdev** [comment]
Results with 64b226a and 0c483e4

First call
--------
50 segments (93ms - 88ms)
100 segments (200ms - 103ms) 
200 segments (250ms - 104ms)

**Comment by darquesdev** [comment]
**SegmentsEntryIds in session**

**Approach**:
- User segments are calculated and persisted in the user session 
- In the following calls the user segments are passed to the method of retrieving segments
- In _isMember_, when it is time to check if a user belongs to a segment based on the model, it simply checks if this segment is among those already calculated / matched.
- User segments have not been separated between model/context because it is not necessary. Simply It's checked that a particular segment is within the already passed ones

**Difficulties**:
- Segment changes are not seen until the user logs out including: Changes in segments, users, organizations and criteria contributors
- There's no way to clear the user session after changing  a model

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:bundle

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by cgoncas** [comment]
ci:test:bundle

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:tango

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/odata/search/ODataSearchAdapterImpl.java [hunk 1]
```diff
@@ -59,28 +64,16 @@ public Hits search(
 			SearchContext searchContext = _createSearchContext(
 				companyId, start, end);
 
-			Query query = _getQuery(
-				filterString, className, entityModel, filterParser, locale,
-				searchContext);
+			Indexer<Organization> indexer = _indexerRegistry.getIndexer(
```
Comment (cgoncas) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/odata/search/ODataSearchAdapterImpl.java:67]: He @darquesdev !!!, there is a typo problem in here with Indexer<Organization> indexer. d you mind changing it and squash it t the original commit?.

Comment (darquesdev) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/odata/search/ODataSearchAdapterImpl.java:67]: sure! I'll do it

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java [hunk 2]
```diff
@@ -141,23 +145,36 @@ public int getSegmentsEntryClassPKsCount(long segmentsEntryId)
 		List<SegmentsEntry> segmentsEntries =
 			segmentsEntryLocalService.getSegmentsEntries(
 				groupId, true, getSource(), className, QueryUtil.ALL_POS,
-				QueryUtil.ALL_POS, null);
+				QueryUtil.ALL_POS,
```
Comment (cgoncas) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java:148]: Would be awesome if we could stop sorting this query.

**Comment by darquesdev** [comment]
Let's move this PR to liferay-tango: https://github.com/liferay-tango/liferay-portal/pull/370

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java (hunk 2 above)
Comment (darquesdev) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java:148]: what do you mean? passing a null value as OrderByComparator?

**Pull request closed without merging**